### ISSUE TITLE
Add an environment variable for velocity secret

### ISSUE THREAD

**Issue by Cubxity** [PaperMC/Paper#6332]
### Is your feature request related to a problem?

When deploying Minecraft servers at scale, you would want to automate as much as you can. What I would like to achieve is an immutable Docker image to deploy my Minecraft server. I would like to store Bukkit, Spigot, Paper, etc config inside the Docker image. However, it is not a good idea to store sensitive information like forwarding secret inside a Docker image. 

### Describe the solution you'd like.

What would be ideal is to have an environment variable that can be defined at runtime, e.g. `VELOCITY_FORWARDING_SECRET`. This makes it easy to set in an environment like Kubernetes or Docker.

### Describe alternatives you've considered.

I have considered templating paper.yml as a ConfigMap with a templating tool like Helm, but this creates an enormous ConfigMap. 

### Agreements

- [X] I have searched for and ensured there isn't already an open issue regarding this.
- [X] I have ensured the feature I'm requesting isn't already in the latest supported Paper build.

**Comment by MiniDigger** [comment]
ideally, we would have a full spring like value system, where every config entry can be overridden with a system property or an environment variable, rather than implementing ven vars manually

**Comment by Proximyst** [comment]
Perhaps moving to pico cli for all config options could help there, alternatively e.g. `-paper-merge-config 'yaml file contents here'` to merge into the YamlConfiguration before we start parsing?

**Comment by deniszitu** [comment]
Store sensitive information like forwarding secret inside a enviroment variable is a bad ideia, the correct thing is to use a file with the right permissions, like [docker secret](https://docs.docker.com/engine/swarm/secrets/).

https://github.com/PaperMC/Velocity/issues/463
